File: project-euler/problem_19/solution.py
from datetime import datetime, timedelta

start_date: datetime = datetime(1901, 1, 1)
end_date: datetime = datetime(2000, 12, 31)
current_date: datetime = start_date
number_of_sundays: int = 0

while current_date <= end_date:
    if current_date.weekday() == 6 and current_date.day == 1:
        number_of_sundays += 1
    current_date += timedelta(days=1)

print(number_of_sundays)

File: project-euler/problem_19/solution.dart
import 'dart:mirrors';

void main() {
  final DateTime startDate = DateTime.parse('1901-01-01');
  final DateTime endDate = DateTime.parse('2000-12-31');
  DateTime currrentDate = startDate;
  int numberOfSundays = 0;

  // While `currentDate` is before or on endDate
  while (currrentDate.compareTo(endDate) <= 0) {
    if ((currrentDate.weekday == 6) && (currrentDate.day == 1)) {
      numberOfSundays += 1;
    }
    currrentDate = currrentDate.add(Duration(days: 1));
  }

  print(numberOfSundays);
}
